### PR TITLE
📦 Transformation Blocks Add to Canvas

### PR DESCRIPTION
**Pull request by arifluthfi16**
Addition:

1. Added TransformationCatalog & TransformationBlock DTO for data handling
2. Created base node handler
3. Created add to canvas feature
4. Created empty modal for the configuration

### CODE REVIEW

**Review by mattigrthr — CHANGES_REQUESTED** [review]
### Keep styling close to designs

I feel we can easily keep the styling closer to the designs. In particular the icon is too big and the name of the transformation block not nicely aligned with small font size and regular (can be bold and bigger as well as aligned to the left).

Design:

image

Current implementation:

image

### Adding data blocks fails

When dragging a data source from the left onto the canvas the block isn't rendered properly.

image

File: kuwala/core/canvas/src/components/Nodes/TransformationBlock/TransformationBlock.js [hunk 1]
```diff
@@ -0,0 +1,98 @@
...
+                        <FontAwesomeIcon
+                            icon={data.transformationBlock.transformationCatalog.icon}
+                            style={{
+                                width: 58,
+                                height: 58,
+                                color: KUWALA_PURPLE,
+                            }}
+                            draggable={false}
+                            className={'user-select-none'}
+                        />
```
[the remark this answers is not in the document]
Comment (arifluthfi16) [reply kuwala/core/canvas/src/components/Nodes/TransformationBlock/TransformationBlock.js:30]: Resolved by creating a common component `Icon`

File: kuwala/core/canvas/src/components/Nodes/TransformationBlock/TransformationBlock.js [hunk 2]
```diff
@@ -0,0 +1,98 @@
...
+
+const TransformationBlock = ({data}) => {
+    const {toggleTransformationConfigModal} = useStoreActions(actions => actions.common);
+    const {toggleDataView, setSelectedElementByTransformationBlockId} = useStoreActions(actions => actions.canvas);
+
+    return (
+        <div
+            className={Classes.BlockWrapper}
+        >
+                {/* BODY */}
```
[the remark this answers is not in the document]
Comment (arifluthfi16) [reply kuwala/core/canvas/src/components/Nodes/TransformationBlock/TransformationBlock.js]: Resolved by creating a general block component

File: kuwala/core/canvas/src/components/Common/Block/index.js [hunk 3]
```diff
@@ -0,0 +1,118 @@
...
+import Classes from "../../Nodes/TransformationBlock/TransformationBlockStyle";
+import Icon from "../Icon";
+import Button from "../Button";
+import {Handle} from "react-flow-renderer";
+import {COLOR_MAP, KUWALA_GRAY, KUWALA_PURPLE} from "../../../constants/styling";
+import React from "react";
+
+const Block = ({
+        blockColor='kuwalaGreen',
+        title='Title',
```
Comment (mattigrthr) [inline kuwala/core/canvas/src/components/Common/Block/index.js:11]: Shouldn't have a default value as we want to make sure it is always passed down to the component.

File: kuwala/core/canvas/src/components/Common/Block/index.js [hunk 4]
```diff
@@ -0,0 +1,118 @@
...
+        primaryButtonActive=true,
+        primaryButtonOnclick,
```
Comment (mattigrthr) [inline kuwala/core/canvas/src/components/Common/Block/index.js:13]: Should be `primaryButtonOnClick`.

File: kuwala/core/canvas/src/components/Common/Block/index.js [hunk 5]
```diff
@@ -0,0 +1,118 @@
...
+        secondaryButtonActive=true,
+        secondaryButtonOnclick,
+        leftHandleActive=true,
+        hideLeftHandle=false,
+        rightHandleActive=true,
+        hideRightHandle=false,
+        data,
+        icon='cogs'
```
Comment (mattigrthr) [inline kuwala/core/canvas/src/components/Common/Block/index.js:21]: Same here, no default value to make sure it's always passed down.

File: kuwala/core/canvas/src/components/Common/Block/index.js [hunk 6]
```diff
@@ -0,0 +1,118 @@
...
+                        borderBottomRightRadius: 0,
+                    }}
+                    d={`${data.id}`}
+                    type="target"
+                    position="left"
+                    isConnectable={leftHandleActive}
+                />
+            )
+        }
+        return <></>
```
Comment (mattigrthr) [inline kuwala/core/canvas/src/components/Common/Block/index.js:47]: If something is not rendered do `return undefined`.

File: kuwala/core/canvas/src/components/Common/Block/index.js [hunk 7]
```diff
@@ -0,0 +1,118 @@
...
+                        borderBottomRightRadius: 12,
+                    }}
+                    d={`${data.id}`}
+                    type="source"
+                    position="right"
+                    isConnectable={rightHandleActive}
+                />
+            )
+        }
+        return <></>
```
Comment (mattigrthr) [inline kuwala/core/canvas/src/components/Common/Block/index.js:72]: Same here

**Comment by mattigrthr** [comment]
### Disabled configure button

The `configure` button of a transformation block should be disabled as long as the block isn't connected to preceding blocks (see designs).

**Review by mattigrthr — APPROVED** [review]

**Pull request merged**